### PR TITLE
Migrate to API definitions to openshift/api

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR migrates API definitions to openshift/api, this includes:
- machine/machineset definitions are imported from openshift/api
- provider spec is removed and imported from openshift/api
- provider spec serialization is moved inside actuator package

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,21 +9,15 @@ require (
...
 	k8s.io/klog/v2 v2.9.0
 	k8s.io/utils v0.0.0-20210819203725-bdf08cb9a70a
 	sigs.k8s.io/controller-runtime v0.9.6
-	sigs.k8s.io/controller-tools v0.6.2
-	sigs.k8s.io/yaml v1.2.0
-)
-
-replace (
-	sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20210622023641-c69a3acaee27
-	sigs.k8s.io/cluster-api-provider-azure => github.com/openshift/cluster-api-provider-azure v0.1.0-alpha.3.0.20210816141152-a7c40345b994
```
Comment (JoelSpeed) [inline go.mod:28]: This makes me happy 😄

File: pkg/actuators/machine/utils.go [hunk 2]
```diff
@@ -406,3 +406,67 @@ func getClusterID(machine *machinev1.Machine) (string, bool) {
...
+	}
+
+	providerStatus := new(machinev1.AWSMachineProviderStatus)
+	if err := json.Unmarshal(rawExtension.Raw, providerStatus); err != nil {
+		return nil, fmt.Errorf("error unmarshalling providerStatus: %v", err)
+	}
+
+	klog.V(5).Infof("Got provider Status from raw extension: %+v", providerStatus)
+	return providerStatus, nil
+}
```
Comment (JoelSpeed) [inline pkg/actuators/machine/utils.go:472]: Do these need to be exported functions?

**Review by JoelSpeed — COMMENTED** [review]
Please add an appropriate PR description and see my question inline, otherwise LGTM

File: pkg/actuators/machine/utils.go (hunk 2 above)
Comment (alexander-demichev) [reply pkg/actuators/machine/utils.go:472]: I think yes, if anyone is importing our fork they might use it.

Comment (JoelSpeed) [reply pkg/actuators/machine/utils.go:472]: If that is a concern, can I suggest that we put these into their own package, don't remove them from the API package but instead "deprecate" them by creating aliases? (We may want to do this for our API types too to make this transition smoother)

```
// Deprecated: Use utils.RawExtensionFromProviderSpec instead.
var RawExtensionFromProviderSpec = utils.RawExtensionFromProviderSpec
```

Comment (alexander-demichev) [reply pkg/actuators/machine/utils.go:472]: We can do aliases for these functions, but API types will get outdated quickly which means that for further use people will have to switch to openshift/api anyway.

**Review by elmiko — COMMENTED** [review]
generally looks good to me
/approve

File: pkg/apis/awsprovider/awsprovider.go [hunk 3]
```diff
@@ -0,0 +1 @@
+package awsprovider
```
Comment (JoelSpeed) [inline pkg/apis/awsprovider/awsprovider.go:1]: Should we not just leave the `group.go` file in place?

File: pkg/apis/awsprovider/v1beta1/conversion.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package v1beta1
+
+import "sigs.k8s.io/cluster-api-provider-aws/pkg/actuators/machine"
+
+var (
+	// RawExtensionFromProviderSpec deprecated: Use machine.RawExtensionFromProviderSpec instead.
```
Comment (JoelSpeed) [inline pkg/apis/awsprovider/v1beta1/conversion.go:6]: I'm not sure this is the right format, there's a particular format in Go that marks it deprecated and causes compiler warnings, can we check that this happens please?

IIRC it has to be on a separate line
```suggestion
	// RawExtensionFromProviderSpec creates a raw version of the given providerSpec.
	// Deprecated: Use machine.RawExtensionFromProviderSpec instead.
```

File: pkg/apis/awsprovider/awsprovider.go (hunk 3 above)
Comment (alexander-demichev) [reply pkg/apis/awsprovider/awsprovider.go:1]: does the file name matter if we keep the package name the same?

Comment (JoelSpeed) [reply pkg/apis/awsprovider/awsprovider.go:1]: Not really, I was just trying to reduce the diff in this PR, this is minor

File: pkg/apis/awsprovider/v1beta1/conversion.go (hunk 4 above)
Comment (alexander-demichev) [reply pkg/apis/awsprovider/v1beta1/conversion.go]: thanks, I've just pushed it

**Pull request merged**